### ISSUE TITLE
Add doc_mapper to the  MergeExecutor.

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-oss/quickwit#2207]
`merge_split_directories` takes a Option<DocMapper> without explainin when and why it is needed.

Would it be possible to add a DocMapper to the `MergeExecutor`?
The `MergePipeline` itself already owns a docmapper.

**Comment by fmassot** [comment]
Good point, seems like the refactoring or the merge pipeline makes things clearer.
